bigbluebutton-html5/imports/ui/components/actions-bar/presentation-options/component.jsx

```diff
@@ -20,20 +21,28 @@ const intlMessages = defineMessages({
   },
 });
 
-const PresentationOptionsContainer = ({ intl, toggleSwapLayout }) => (
-  <Button
-    className={styles.button}
-    icon="presentation"
-    label={intl.formatMessage(intlMessages.restorePresentationLabel)}
-    description={intl.formatMessage(intlMessages.restorePresentationDesc)}
-    color="primary"
-    hideLabel
-    circle
-    size="lg"
-    onClick={toggleSwapLayout}
-    id="restore-presentation"
-  />
-);
+const shouldUnswapLayout = () => {
+  return MediaService.shouldShowScreenshare() || MediaService.shouldShowExternalVideo();
```

Comment:
From the code it looks like we don't handle the case where screenshare is disabled in Data Savings

Reply:
This means if an user has data saving enabled he is not supposed to watch the screenshare?

Reply:
I think if we have data savings disallowing viewing of screenshare, we should be showing the presentation area